nativelib/src/main/scala/scala/scalanative/posix/regex.scala

```diff
@@ -0,0 +1,18 @@
+package scala.scalanative
+package posix
+
+import native._
+
+@extern
+object regex {
+
+  @name("regcomp")
+  def compile(regex: Ptr[CInt], str: CString, num: CInt): CInt = extern
+
+  @name("regexec")
+  def execute(regex: Ptr[CInt],
+              str: CString,
+              num: CInt,
+              ptr: Ptr[CInt],
+              num2: CInt): CInt = extern
+}
```

Comment:
I want to keep the original work from https://github.com/scala-native/scala-native/pull/588/commits/ed537c1fb8fa4b363d5a14ea5241a7ca80a9d3c1